clang/lib/Basic/DiagnosticIDs.cpp

```diff
@@ -575,25 +575,42 @@ DiagnosticIDs::getDiagnosticSeverity(unsigned DiagID, SourceLocation Loc,
       DiagID != diag::fatal_too_many_errors && Diag.FatalsAsError)
     Result = diag::Severity::Error;
 
+  // Rest of the mappings are only applicable for diagnostics associated with a
+  // SourceLocation, bail out early for others.
+  if (!Diag.hasSourceManager())
+    return Result;
+
+  const auto &SM = Diag.getSourceManager();
```

Comment:
This is typically called SrcMgr and typically we keep the full name and don't use auto.

Reply:
hmm not sure where that's coming from, but in my experience it's actually much more common to name this as `SM` and use `auto`. so i'd rather keep it as-is (happy to provide some pointers offline if need be).

Reply:
"SM" is common.
"auto" not that common.
"SourceMgr" is also relatively common, much more readable and fits better to the style guide.

Reply:
`SM` is "fine" (a more descriptive name is better of course), but generally we only use `auto` when the type is spelled out in the assignment. In this case, `getSourceManager()` is pretty close to spelling out the type name and so I'm not really opposed to `auto`.